### DISCUSSION TITLE
Compiler or Code Editor

### DISCUSSION THREAD

**Discussion by RahulAnand442001** [hackclubiter/onEditor#11] [Q&A] [answered]
Just for clarification, can I know if we are going to make a compiler from scratch or just a code editor which will accept the code and give output?

**Comment by saswatsamal** [comment, marked as answer]
Yes. We will make an **Online IDE**.

**Comment by Saswat7101** [comment]
I think first we should go for code editor and then design a compiler from scratch coz for designing a compiler from scratch we have to read a hell lot of things

**Reply by LoopGlitch26** [reply]
Sure, we can do that, as a testing level implementation.

**Reply by Saswat7101** [reply]
Yes yes for testing only

**Comment by LoopGlitch26** [comment]
Yes, @RahulAnand442001 we'll try to make an online compiler. 

We'll also try to focus on the backend and not just the frontend. My point is that, if we make an online code editor we can use **pre-existing** APIs to integrate with the front-end. That'll not include many backends. But if we make our own API and integrate it with our own frontend (which will be the code editor) it'll be a **full-stack** project, and we can have options of going forward with this project into various scopes in the future, such as, production, paper publication, open-source project, etc.  

But we will do everything step-by-step, just like a major project, where we'll integrate many parts of the project into one. Let's see how it goes.

Please, others also give your opinions.

**Comment by RahulAnand442001** [comment]
Ok As you all are saying compilers. So as far as I have researched we need to convert the code into AST and then convert it to machine-readable language. And for each language, there will be separate language-specific compiler codes. Also to run those compilers online we have to deploy the compilers in some cloud instances and containers like Docker. Further using Nginx and Kubernetes we have to maintain the CI/CD pipeline. Ansible will also play a major role in this project. Attached to that we have to think of containerization and orchestration in the cloud. That will be our project entry point.

**Reply by LoopGlitch26** [reply]
Alright, and first we can try to do the compiler for one language.

**Reply by saswatsamal** [reply]
Yes. We'll focus on that only.
